source/misc/bash_startup.in

```diff
@@ -1,257 +1,95 @@
-      else
-          local this_command="";
-      fi;
-
-      # If none of the previous checks have earlied out of this function, then
-      # the command is in fact interactive and we should invoke the user's
-      # preexec hook with the running command as an argument.
-      preexec "$this_command";
-  }
-
-  # Execute this to set up preexec and precmd execution.
-  function iterm2_preexec_install () {
-
-      # *BOTH* of these options need to be set for the DEBUG trap to be invoked
-      # in ( ) subshells.  This smells like a bug in bash to me.  The null stackederr
-      # redirections are to quiet errors on bash2.05 (i.e. OSX's default shell)
-      # where the options can't be set, and it's impossible to inherit the trap
-      # into subshells.
-
-      set -o functrace > /dev/null 2>&1
-      shopt -s extdebug > /dev/null 2>&1
-
-      # Finally, install the actual traps.
-      if ( [ x"$PROMPT_COMMAND" = x ]); then
-        PROMPT_COMMAND="iterm2_preexec_invoke_cmd";
-      else
-        # If there's a trailing semicolon folowed by spaces, remove it (issue 3358).
-        PROMPT_COMMAND="$(echo -n $PROMPT_COMMAND | sed -e 's/; *$//'); iterm2_preexec_invoke_cmd";
-      fi
-      trap 'iterm2_preexec_invoke_exec' DEBUG;
-  }
-
-  # -- begin iTerm2 customization
-
-  function iterm2_begin_osc {
-    printf "\033]"
-  }
-
-  function iterm2_end_osc {
-    printf "\007"
-  }
-
-  # Runs after interactively edited command but before execution
-  function preexec() {
-    iterm2_begin_osc
-    printf "133;C"
-    iterm2_end_osc
-  }
-
-  function iterm2_print_state_data() {
-    iterm2_begin_osc
-    printf "1337;RemoteHost=%s@%s" "$USER" "$iterm2_hostname"
-    iterm2_end_osc
-
-    iterm2_begin_osc
-    printf "1337;CurrentDir=%s" "$PWD"
-    iterm2_end_osc
-
-    iterm2_print_user_vars
-  }
-
-  # Usage: iterm2_set_user_var key value
-  function iterm2_set_user_var() {
-    iterm2_begin_osc
-    printf "1337;SetUserVar=%s=%s" "$1" $(printf "%s" "$2" | base64)
-    iterm2_end_osc
-  }
-
-  # Users can write their own version of this method. It should call
-  # iterm2_set_user_var but not produce any other output.
-  function iterm2_print_user_vars() {
-    true
-  }
-
-  function iterm2_prompt_prefix() {
-    iterm2_begin_osc
-    printf "133;D;\$?"
-    iterm2_end_osc
-
-    iterm2_print_state_data
-
-    iterm2_begin_osc
-    printf "133;A"
-    iterm2_end_osc
-  }
-
-  function iterm2_prompt_suffix() {
-    iterm2_begin_osc
-    printf "133;B"
-    iterm2_end_osc
-  }
-
-  function iterm2_print_version_number() {
-    iterm2_begin_osc
-    printf "1337;ShellIntegrationVersion=1"
-    iterm2_end_osc
-  }
-
-
-  # If hostname -f is slow on your system, set iterm2_hostname before sourcing this script.
-  if [ -z "$iterm2_hostname" ]; then
-    iterm2_hostname=$(hostname -f)
-  fi
-  iterm2_preexec_install
-
-  # This is necessary so the first command line will have a hostname and current directory.
+iterm2_prompt_prefix () {
```

Comment:
For consistency with functions above, this should be named iterm2_print_prompt_prefix